source/js/nav-newsletter.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import ReactGA from "react-ga";
+import ReactDOM from "react-dom";
+import JoinUs from "./components/join/join.jsx";
+
+let foundationSiteURL, csrfToken;
+let isShown = false;
+let primaryNav = document.querySelector(`#primary-nav-container`);
+let narrowMenuContainer = primaryNav.querySelector(
+  ".narrow-screen-menu-container"
+);
+let wideMenuContainer = primaryNav.querySelector(".wide-screen-menu-container");
+let container = document.querySelector("#nav-newsletter-form-wrapper");
+let buttonDesktop = wideMenuContainer.querySelector(".btn-newsletter");
+let buttonMobile = narrowMenuContainer.querySelector(".btn-newsletter");
+let joinUs = container.querySelector(`.join-us.on-nav`);
+let buttonDismiss = container.querySelector(".form-dismiss");
+
+let navNewsletter = {
+  getShownState: () => {
+    return isShown;
+  },
+
+  // Inject a new <JoinUs> component each time so we can show <JoinUs> with its initial state
```

Comment:
What does this comment mean with "each time"? It feels like it's not being detailed enough. If the idea is to "replace what is on the page" with a new JoinUs compnent just to reset its state, then this should probably be changed to something like this:
```
class JoinUs extends Component {
  constructor(props) {
    super(props);
    this.state = this.getInitialState();
  } 
  reset() {
    this.setState(this.getInitialState());
  }
  getInitialState() {
    return { ... };
  }
```
so that we can call the JoinUs component's `.reset()` function without having to replace it with an entirely new component.

Reply:
@Pomax we can't call a React component's method from outside of the React world. That's why I had to unmount the component in `nav-newsletter.js` instead.

Reply:
By each time I mean each time the newsletter section is revealed. I can update the comment.